r-package/grf/R/input_utilities.R

```diff
@@ -204,13 +216,18 @@ create_data_matrices <- function(X, outcome = NULL, treatment = NULL,
   }
 
   if (inherits(X, "dgCMatrix") && ncol(X) > 1) {
-    sparse.data <- cbind(X, outcome, treatment, instrument, sample.weights)
+    sparse.data <- cbind(X, outcome, treatment, instrument, censor, sample.weights)
   } else {
     X <- as.matrix(X)
-    default.data <- as.matrix(cbind(X, outcome, treatment, instrument, sample.weights))
+    default.data <- as.matrix(cbind(X, outcome, treatment, instrument, censor, sample.weights))
+  }
+  if (train.data) {
```

Comment:
Could you explain this change? (I'm not up-to-speed on the survival forest algorithm and may be missing some context).

Reply:
This is only a convince for the `do.call.rcpp` command: 

https://github.com/erikcs/grf/blob/add-survival-forest/r-package/grf/R/survival_forest.R#L202

A follow-up can simplify the other wrappers as well:

https://github.com/grf-labs/grf/blob/master/r-package/grf/R/causal_forest.R#L399

Reply:
Ah, I see now -- that's in fact simpler. To me it seems clearer if we created a new method specifically for test data like `create_test_matrices` (and renamed this to `create_train_matrices`). The new method could skip the logic related to observations. It looks like very little code would be duplicated.

Reply:
Thanks, that was cleaner, I can add `do.call.rcpp` in the other predicts in another PR!